src/app/Marine2/components/boxes/Battery/Battery.tsx

```diff
@@ -29,24 +46,31 @@ const Battery = ({ battery, mode = "compact" }: Props) => {
   }
 
   return (
-    <Box icon={batteryStateIconFormatter(battery.state)} title={battery.name} className="truncate">
+    <Box
+      icon={batteryStateIconFormatter(battery.state)}
+      title={battery.name}
+      className="truncate"
+      getBoxSizeCallback={setBoxSize}
+    >
```

Comment:
I prefer setting the ref on the box right here, and using the `useComponentSize` hook in this file. That way no callback is needed, simplifying the code.

Reply:
I also wanted to do it like this but unfortunately, refs don't really work the same in functional components. 
<img width="787" alt="image" src="https://user-images.githubusercontent.com/119003472/221857765-61a9f736-d14c-4552-bafc-0854147bb3ca.png">